### PR TITLE
[LTS] Catch up LTS azure directory

### PR DESCRIPTION
**Pull request by scottn12**
## Description

This PR catches the azure packages to the correct state in the LTS branch. To do so, we will catch up to commit https://github.com/microsoft/FluidFramework/commit/7fa7dd9ed3f38919d163769a3d22b69fa3218f1a, since this is the last commit before azure packages are moved into the client release group (commit https://github.com/microsoft/FluidFramework/commit/16838fa843278a646fb842533ff23fd7635049b9).

## Misc
[AB#4214](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4214)

### CODE REVIEW

File: build-tools/packages/build-tools/data/layerInfo.json
Comment (scottn12) [inline build-tools/packages/build-tools/data/layerInfo.json:1]: This file was edited to account for the `azure-local-service` package being moved from the `server` directory to the `azure` directory. Similarly, the changes made in `package.json` and `tools/pipelines/build-azure-local-service.yml` were to account for this package moving as well.

File: tools/pipelines/build-azure.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ parameters:
 - name: buildToolsVersionToInstall
   displayName: Fluid build tools version (default = installs version in repo)
   type: string
-  default: latest
+  default: "^0.4.4000"
```
Comment (tylerbutler) [inline tools/pipelines/build-azure.yml:28]: Should this version be different from the others?

Comment (scottn12) [reply tools/pipelines/build-azure.yml:28]: This is the version that azure packages are using in the release/azure/1.1 branch. Although I think we could upgrade the version to match other packages I don't think it's necessary

File: packages/framework/request-handler/src/test/requestHandlers.spec.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ describe("RequestParser", () => {
             const requestParser = RequestParser.create({ url: "/objectId/route", headers: { wait: true } });
             const response = await rootDataStoreRequestHandler(requestParser, runtime);
             assert.equal(response?.status, 200);
-            assert.equal(response.value.route, "/route");
+            assert.equal(response?.value.route, "/route");
```
Comment (ssimic2) [inline packages/framework/request-handler/src/test/requestHandlers.spec.ts:96]: curious why this change as part of this PR?

File: tools/pipelines/build-client.yml [hunk 3]
```diff
@@ -25,7 +25,7 @@ parameters:
 - name: buildToolsVersionToInstall
   displayName: Fluid build tools version (default = installs version in repo)
   type: string
-  default: latest
+  default: "^0.6.0-110101"
```
Comment (ssimic2) [inline tools/pipelines/build-client.yml:28]: If the scope is azure only, curious why this change here?

File: tools/pipelines/build-build-tools.yml [hunk 4]
```diff
@@ -24,7 +24,7 @@ parameters:
 - name: buildToolsVersionToInstall
   displayName: Fluid build tools version (default = installs version in repo)
   type: string
-  default: latest
+  default: "^0.6.0-110101"
```
Comment (ssimic2) [inline tools/pipelines/build-build-tools.yml:27]: You see to be using different version for different pipelines. Please sync with @tylerbutler and align on the action before merging this PR.

**Review by ssimic2 — APPROVED** [review]

File: packages/framework/request-handler/src/test/requestHandlers.spec.ts (hunk 2 above)
Comment (scottn12) [reply packages/framework/request-handler/src/test/requestHandlers.spec.ts:96]: I'm not sure exactly why, but for whatever reason the linter gave me trouble for this one line

File: tools/pipelines/build-client.yml (hunk 3 above)
Comment (scottn12) [reply tools/pipelines/build-client.yml:28]: There is a bug with the latest version of build tools that was affecting this repo. @tylerbutler suggested pinning the version to avoid the issue

**Comment by scottn12** [comment]
Closing in favor of https://github.com/microsoft/FluidFramework/pull/15909
